Generate settings as JSON file

mfine opened the pull request:
Simple utility to generate our settings as a JSON file for consumption by other tooling.

Generated JSON looks like:

```json
{
  "ext_events": {
    "edge_trigger": "None"
  },
  "float_kf": {
    "amb_init_var": "1e+25",
    "code_var": "40000",
    "new_amb_var": "1e+25",
    "phase_var": "0.0144"
  },
  "frontend": {
    "antenna_selection": "External"
  },
  "iar": {
    "code_var": "40000",
    "phase_var": "0.0144"
  },
  "sbp": {
    "obs_msg_max_size": "104"
  },
  "simulator": {
    "base_ecef_x": "-2700303.10144",
    "base_ecef_y": "-4292474.39651",
    "base_ecef_z": "3855434.34087",
    "cn0_sigma": "0.300000011921",
    "enabled": "False",
    "mode_mask": "15",
    "num_sats": "9",
    "phase_sigma": "0.0299999993294",
    "pos_sigma": "1.5",
    "pseudorange_sigma": "4",
    "radius": "100",
    "speed": "4",
    "speed_sigma": "0.15000000596"
  },
  "solution": {
    "dgnss_filter": "Fixed",
    "dgnss_solution_mode": "Low Latency",
    "known_baseline_d": "0",
    "known_baseline_e": "0",
    "known_baseline_n": "0",
    "output_every_n_obs": "2",
    "soln_freq": "10"
  },
  "surveyed_position": {
    "broadcast": "False",
    "surveyed_alt": "0",
    "surveyed_lat": "0",
    "surveyed_lon": "0"
  },
  "system_info": {
    "firmware_built": "Jun 16 2015 00:15:21",
    "firmware_version": "v0.17",
    "hw_revision": "piksi_2.3.1",
    "nap_channels": "11",
    "nap_fft_index_bits": "13",
    "nap_version": "v0.13",
    "serial_number": "36828"
  },
  "system_monitor": {
    "heartbeat_period_milliseconds": "1000",
    "watchdog": "True"
  },
  "telemetry_radio": {
    "configuration_string": "AT&F,ATS1=115,ATS2=128,ATS5=0,AT&W,ATZ"
  },
  "track": {
    "iq_output_mask": "0"
  },
  "uart_ftdi": {
    "baudrate": "1000000",
    "mode": "SBP",
    "sbp_message_mask": "65535"
  },
  "uart_uarta": {
    "baudrate": "115200",
    "configure_telemetry_radio_on_boot": "True",
    "mode": "SBP",
    "sbp_message_mask": "64"
  },
  "uart_uartb": {
    "baudrate": "115200",
    "configure_telemetry_radio_on_boot": "True",
    "mode": "SBP",
    "sbp_message_mask": "65280"
  }
}
```

All strings because there's no type information.

/cc @fnoble @mookerji @denniszollo

mookerji commented on piksi_tools/settings.py:
```diff
@@ -0,0 +1,87 @@
+  Get and parse arguments.
+  """
+  import argparse
+  parser = argparse.ArgumentParser(description='Acquisition Monitor')
```
Not the acquisition monitor.

mookerji commented:
JSON (instead of YAML) aside, and the nit aside, looks good to me. I imagine we'll replace this with more stable settings messages?

mfine commented:
I moved this over to a diagnostics generators and write everything out in yaml (see below). Did not add any ID's in there - do we want them in there?

```yaml
settings:
  ext_events:
    edge_trigger: None
  float_kf:
    amb_init_var: 1e+25
    code_var: '40000'
    new_amb_var: 1e+25
    phase_var: '0.0144'
  frontend:
    antenna_selection: External
  iar:
    code_var: '40000'
    phase_var: '0.0144'
  sbp:
    obs_msg_max_size: '104'
  simulator:
    base_ecef_x: '-2700303.10144'
    base_ecef_y: '-4292474.39651'
    base_ecef_z: '3855434.34087'
    cn0_sigma: '0.300000011921'
    enabled: 'False'
    mode_mask: '15'
    num_sats: '9'
    phase_sigma: '0.0299999993294'
    pos_sigma: '1.5'
    pseudorange_sigma: '4'
    radius: '100'
    speed: '4'
    speed_sigma: '0.15000000596'
  solution:
    dgnss_filter: Fixed
    dgnss_solution_mode: Low Latency
    known_baseline_d: '0'
    known_baseline_e: '0'
    known_baseline_n: '0'
    output_every_n_obs: '2'
    soln_freq: '10'
  surveyed_position:
    broadcast: 'False'
    surveyed_alt: '0'
    surveyed_lat: '0'
    surveyed_lon: '0'
  system_info:
    firmware_built: Jun 18 2015 12:45:28
    firmware_version: v0.17-27-ge2c1aac
    hw_revision: piksi_2.3.1
    nap_channels: '11'
    nap_fft_index_bits: '13'
    nap_version: v0.13
    serial_number: '36828'
  system_monitor:
    heartbeat_period_milliseconds: '1000'
    watchdog: 'True'
  telemetry_radio:
    configuration_string: AT&F,ATS1=115,ATS2=128,ATS5=0,AT&W,ATZ
  track:
    iq_output_mask: '0'
  uart_ftdi:
    baudrate: '1000000'
    mode: SBP
    sbp_message_mask: '65535'
  uart_uarta:
    baudrate: '115200'
    configure_telemetry_radio_on_boot: 'True'
    mode: SBP
    sbp_message_mask: '64'
  uart_uartb:
    baudrate: '115200'
    configure_telemetry_radio_on_boot: 'True'
    mode: SBP
    sbp_message_mask: '65280'
versions:
  bootloader: v1.2
  sbp: 0
```

mookerji commented:
Other IDs, like the NAP? I think this is fine for the being. My only nit would be coercing the obviously numeric values from strings to floats/ints or something.

mfine commented:
Yeah, but it's kind of hard to do without a schema or something - kind of have to do the amazon thing and treat everything as a string and coerce them on consumption. Better idea?

mookerji commented:
Yeah, that works for me. Feel free to merge whenever.

mfine commented:
Bringing this in.
